lib/output.js

```diff
@@ -74,9 +74,12 @@ function OutputStream(options) {
     var OUTPUT = "";
 
     function to_ascii(str, identifier) {
-        return str.replace(/[\u0000-\u001f\u007f-\uffff]/g, function(ch) {
-            var code = ch.charCodeAt(0).toString(16);
-            if (code.length <= 2 && !identifier) {
+        return str.replace(/[\ud800-\udbff][\udc00-\udfff]|[\u0000-\u001f\u007f-\uffff]/g, function(ch) {
```

Comment:
Adding full unicode support. Also the identifier tokenizer on master didn't account surrogates. But this patch is just including surrogated identifier chars to identifiers.

The consequence is that parsing identifiers and strings have to be adopted to it, because they are the only pieces of code that can "play" with surrogates.

Keywords, punctuators, etc... don't use surrogates, so there is no change needed there. Comments are just read and parsed.

With extended unicode support, we add the ability to process more characters, we can handle the new unicode escape sequence without problems. It's just adding better support for unicode.